pkg/asset/machines/aws/machines.go

```diff
@@ -95,20 +98,37 @@ func provider(clusterID string, platform *aws.Platform, mpool *aws.MachinePool,
 		IAMInstanceProfile: &awsprovider.AWSResourceReference{ID: pointer.StringPtr(fmt.Sprintf("%s-%s-profile", clusterID, role))},
 		UserDataSecret:     &corev1.LocalObjectReference{Name: userDataSecret},
 		CredentialsSecret:  &corev1.LocalObjectReference{Name: "aws-cloud-credentials"},
-		Subnet: awsprovider.AWSResourceReference{
-			Filters: []awsprovider.Filter{{
-				Name:   "tag:Name",
-				Values: []string{fmt.Sprintf("%s-private-%s", clusterID, az)},
-			}},
-		},
-		Placement: awsprovider.Placement{Region: platform.Region, AvailabilityZone: az},
+		Placement:          awsprovider.Placement{Region: platform.Region, AvailabilityZone: az},
 		SecurityGroups: []awsprovider.AWSResourceReference{{
 			Filters: []awsprovider.Filter{{
 				Name:   "tag:Name",
 				Values: []string{fmt.Sprintf("%s-%s-sg", clusterID, role)},
 			}},
 		}},
-	}, nil
+	}
+
+	if len(platform.Subnets) > 0 {
+		metas, err := awsconfig.SubnetMetadata(context.TODO(), platform.Subnets)
+		if err != nil {
+			return nil, err
+		}
+		for id, meta := range metas {
+			if meta.Zone == az && meta.Public == false {
+				config.Subnet.ID = pointer.StringPtr(id)
+				break
+			}
+		}
+		if config.Subnet.ID == nil {
+			return nil, errors.Wrapf(err, "no private subnet found in %s: %#+v", az, metas)
```

Comment:
`%#+v` `metas` is too verbose.
maybe, `no private subnet found for zone {} in the subnets provided {}`

Reply:
> `%#+v` `metas` is too verbose.

I want it to be obvious why each subnet was rejected.  Added a `rejected` slice with the reasons for a more compact error with 951bf998b -> eb6a6dbb6.